test/library/olm/olm.go

```
@@ -0,0 +1,366 @@
+	return wait.PollUntilContextTimeout(ctx, 10*time.Second, 3*time.Minute, true, func(ctx context.Context) (bool, error) {
+		csv, err := dynamicClient.Resource(GetCSVGVR()).Namespace(namespace).Get(ctx, csvName, metav1.GetOptions{})
+		if err != nil {
+			klog.Warningf("Failed to get CSV %s: %v", csvName, err)
+			return false, nil
+		}
+
+		// Get the phase from status
+		phase, found, err := unstructured.NestedString(csv.Object, "status", "phase")
+		if err != nil || !found {
+			klog.Warningf("CSV %s has no phase yet", csvName)
+			return false, nil
```

golangci-lint (2.12.2)

[error] error is not nil (line 285) but it returns nil

(nilerr)